pycaret/containers/models/time_series.py

```diff
@@ -1972,6 +1972,97 @@ def _set_tune_distributions(self) -> Dict[str, List[Any]]:
         return tune_distributions
 
 
+class CatBoostCdsDtContainer(CdsDtContainer):
+    id = "catboost_cds_dt"
+    name = "CatBoost Regressor w/ Cond. Deseasonalize & Detrending"
+    active = True  # set back to True as the parent has False
+
+    def __init__(self, globals_dict: dict) -> None:
+        # suppress output
+        logging.getLogger("catboost").setLevel(logging.ERROR)
+
+        self.use_gpu = globals_dict["gpu_param"] == "force" or (
+            globals_dict["gpu_param"] and len(globals_dict["X_train"]) >= 50000
+        )
+
+        super().__init__(globals_dict=globals_dict)
+
+    def return_model_class(self):
+        try:
+            import catboost
+        except ImportError:
+            self.logger.warning("Couldn't import catboost.CatBoostRegressor")
+            self.active = False
+            return
+
+        catboost_version = tuple([int(x) for x in catboost.__version__.split(".")])
+        if catboost_version < (0, 23, 2):
```

Comment:
We can read the required catboost version from the `requirements-optional.txt` file, otherwise it could be hard to remember to update this part of the code when updating the requirements file.

Reply:
But the `requirements.txt` would reside outside the module and wont be accessible from within - correct?

Reply:
Checking it like it's being done in here is correct, as the user may choose to install catboost separately.